mmv1/third_party/terraform/services/secretmanagerregional/data_source_secret_manager_regional_secrets.go

```diff
@@ -0,0 +1,176 @@
+package secretmanagerregional
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+func DataSourceSecretManagerRegionalRegionalSecrets() *schema.Resource {
+	dsSchema := tpgresource.DatasourceSchemaFromResourceSchema(ResourceSecretManagerRegionalRegionalSecret().Schema)
+	return &schema.Resource{
+		Read: dataSourceSecretManagerRegionalRegionalSecretsRead,
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"location": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"filter": {
+				Type: schema.TypeString,
+				Description: `Filter string, adhering to the rules in List-operation filtering (https://cloud.google.com/secret-manager/docs/filtering).
+List only secrets matching the filter. If filter is empty, all regional secrets are listed from the specified location.`,
+				Optional: true,
+			},
+			"secrets": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: dsSchema,
+				},
+			},
+		},
+	}
+}
+
+func dataSourceSecretManagerRegionalRegionalSecretsRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*transport_tpg.Config)
+	userAgent, err := tpgresource.GenerateUserAgentString(d, config.UserAgent)
+	if err != nil {
+		return err
+	}
+
+	url, err := tpgresource.ReplaceVars(d, config, "{{SecretManagerRegionalBasePath}}projects/{{project}}/locations/{{location}}/secrets")
```

Comment:
Can we not use the entire path of the resource from config here.
We seem to be using : SecretManagerRegionalBasePath here.
Can we not use the base_url here?

Reply:
Internally, it uses the base_url defined in the Product.yaml. The entry for `SecretManagerRegionalBasePath` is generated in the `transport/config.go` file which be found here: https://github.com/hashicorp/terraform-provider-google/blob/main/google/transport/config.go#L563